mesonbuild/build.py

```diff
@@ -1735,6 +1738,25 @@ def validate_cross_install(self, environment):
         pass
 
 
+class CustomTargetIndex:
+
+    """A special opaque object returned by indexing a CustomTaget. This object
+    exists in meson, but acts as a proxy in the backends, making targets depend
+    on the CustomTarget it's derived from, but only adding one source file to
+    the sources.
+    """
+
+    def __init__(self, target, output):
+        self.target = target
+        self.output = output
```

Comment:
This also needs a `__repr__` implementation because we sometimes print internal meson objects in exceptions and such, and it's good to have a debuggable representation then.

Reply:
can do.